### ISSUE TITLE
Release 4.0.0-beta001

### ISSUE THREAD

**Issue by blairconrad** [FakeItEasy/FakeItEasy#1165]
Can be labelled **ready** when all other issues on this milestone are closed.

- [x] ~~run code analysis in VS in *Release* mode and address all coding and style violations (send a regular PR which must be merged before continuing)~~
- [x] ensure completeness of draft [release notes](https://github.com/FakeItEasy/FakeItEasy/releases), including non-owner contributors, if any (move release notes forward from any pre-releases to the current release)
- change "vNext" names to appropriate x.y.z version for the next:
    - [x] GitHub Release
    - [x] release checklist (this issue)
- create new GitHub artifacts for the next release:
    - [x] create a new draft GitHub Release with name "vNext"
    - [x] create a new issue (like this one) with name "Release vNext" for the next release, adding it to the new milestone
- [x] tag develop with x.y.z and push to upstream. This will push the nupkgs to [nuget.org](https://www.nuget.org/profiles/FakeItEasy)
- [x] de-list pre-release or superseded buggy NuGet packages if present
- [x] attach all nupkgs from the [tagged build](https://ci.appveyor.com/project/FakeItEasy/fakeiteasy/history) to the GitHub Release
- [x] publish the GitHub release
- [x] tweet, mentioning contributors, and post link as comment here for easy retweeting ;-)
- [x] post tweet in [Gitter](https://gitter.im/FakeItEasy/FakeItEasy)
- [x] ~~post a link to the GitHub Release in each issue in this milestone, with thanks to contributors~~
- [ ] close this issue

**Comment by blairconrad** [comment]
https://twitter.com/fakeiteasyfx/status/884022063345926144

https://twitter.com/fakeiteasyfx/status/884022188969537537
